File: CS/DashboardNextPrevTab/ViewerForm.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardWin;
using DevExpress.XtraEditors;
using System.Linq;
using System.Windows.Forms;

namespace DashboardNextPrevTab
{
    public partial class ViewerForm : XtraForm {
        const int slideShowIntervalInMilliseconds = 3000;
        Timer slideShowTimer = new Timer();
        TabContainerDashboardItem TabContainer {
            get {
                Dashboard dashboard = dashboardViewer.Dashboard;
                return dashboard.Items.SingleOrDefault(i => i is TabContainerDashboardItem) as TabContainerDashboardItem;
            }
        }
        public ViewerForm() {
            InitializeComponent();

            dashboardViewer.CustomizeDashboardTitle += DashboardViewerCustomizeDashboardTitle;
            dashboardViewer.SelectedTabPageChanged += DashboardViewerSelectedTabPageChanged;

            SetUpTimer();
        }
        void SetUpTimer() {
            slideShowTimer.Interval = slideShowIntervalInMilliseconds;
            slideShowTimer.Tick += SlideShowTimerTick;
        }
        void DashboardViewerCustomizeDashboardTitle(object sender, CustomizeDashboardTitleEventArgs e) {
            DashboardToolbarItem nextTabItem = new DashboardToolbarItem("Next tab", NextPrevTabItemClick);
            nextTabItem.Tag = NextPrevValue.Next;
            nextTabItem.SvgImage = imageCollection["Next"];
            e.Items.Insert(0, nextTabItem);

            DashboardToolbarItem prevTabItem = new DashboardToolbarItem("Previous tab", NextPrevTabItemClick);
            prevTabItem.Tag = NextPrevValue.Prev;
            prevTabItem.SvgImage = imageCollection["Prev"];
            e.Items.Insert(0, prevTabItem);

            DashboardToolbarItem showTabHederItem = new DashboardToolbarItem(TabContainer.ShowCaption, "Show tab headers", ShowHideTabHeadersItemClick);
            showTabHederItem.SvgImage = imageCollection["ShowCaption"];
            e.Items.Insert(0, showTabHederItem);

            DashboardToolbarItem slideShowItem = new DashboardToolbarItem(slideShowTimer.Enabled, "Slideshow", SlideShowItemClick);
            slideShowItem.SvgImage = imageCollection["Slideshow"];
            e.Items.Insert(0, slideShowItem);
        }
        void SlideShowTimerTick(object sender, System.EventArgs e) {
            transitionManager.StartTransition(dashboardViewer);
            try {
                ShowNextPrevTab(NextPrevValue.Next);
            } finally {
                transitionManager.EndTransition();
            }
        }
        void ShowNextPrevTab(NextPrevValue value) {
            TabContainerDashboardItem tabContainer = TabContainer;
            if(tabContainer != null) {
                int increment = value == NextPrevValue.Next ? 1 : -1;
                string tabContainerName = tabContainer.ComponentName;
                int selectedIndex = dashboardViewer.GetSelectedTabPageIndex(tabContainerName);
                int pageCount = tabContainer.TabPages.Count;
                dashboardViewer.SetSelectedTabPage(tabContainerName, (selectedIndex + pageCount + increment) % pageCount);
            }
        }
        void NextPrevTabItemClick(DashboardToolbarItemClickEventArgs args) {
            ShowNextPrevTab((NextPrevValue)args.Item.Tag);
        }
        void SlideShowItemClick(DashboardToolbarItemClickEventArgs args) {
            if(args.Item.Checked.HasValue && args.Item.Checked.Value)
                slideShowTimer.Start();
            else
                slideShowTimer.Stop();
        }
        void ShowHideTabHeadersItemClick(DashboardToolbarItemClickEventArgs args) {
            TabContainerDashboardItem tabContainer = TabContainer;
            if(tabContainer != null) {
                if(args.Item.Checked.HasValue && args.Item.Checked.Value) {
                    tabContainer.ShowCaption = true;
                } else {
                    tabContainer.ShowCaption = false;
                }
            }
        }
        void DashboardViewerSelectedTabPageChanged(object sender, SelectedTabPageChangedEventArgs e) {
            TabContainerDashboardItem tabContainer = TabContainer;
            if(tabContainer != null && !tabContainer.ShowCaption) {
                string selectedTabPageName = e.Page;
                dashboardViewer.Dashboard.Title.Text = TabContainer.TabPages[selectedTabPageName].Name;
            }
        }
        protected override void Dispose(bool disposing) {
            if(disposing && (components != null)) {
                components.Dispose();
            }
            if(disposing) {
                slideShowTimer.Tick -= SlideShowTimerTick;
                slideShowTimer.Dispose();
            }
            base.Dispose(disposing);
        }
    }
    enum NextPrevValue {
        Next,
        Prev
    }
}

File: VB/DashboardNextPrevTab/ViewerForm.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardWin
Imports DevExpress.XtraEditors
Imports System.Linq
Imports System.Windows.Forms

Namespace DashboardNextPrevTab

    Public Partial Class ViewerForm
        Inherits XtraForm

        Const slideShowIntervalInMilliseconds As Integer = 3000

        Private slideShowTimer As Timer = New Timer()

        Private ReadOnly Property TabContainer As TabContainerDashboardItem
            Get
                Dim dashboard As Dashboard = dashboardViewer.Dashboard
                Return TryCast(dashboard.Items.SingleOrDefault(Function(i) TypeOf i Is TabContainerDashboardItem), TabContainerDashboardItem)
            End Get
        End Property

        Public Sub New()
            InitializeComponent()
            AddHandler dashboardViewer.CustomizeDashboardTitle, AddressOf DashboardViewerCustomizeDashboardTitle
            AddHandler dashboardViewer.SelectedTabPageChanged, AddressOf DashboardViewerSelectedTabPageChanged
            dashboardViewer.UpdateDashboardTitle()
            SetUpTimer()
        End Sub

        Private Sub SetUpTimer()
            slideShowTimer.Interval = slideShowIntervalInMilliseconds
            AddHandler slideShowTimer.Tick, AddressOf SlideShowTimerTick
        End Sub

        Private Sub DashboardViewerCustomizeDashboardTitle(ByVal sender As Object, ByVal e As CustomizeDashboardTitleEventArgs)
            Dim nextTabItem As DashboardToolbarItem = New DashboardToolbarItem("Next tab", AddressOf NextPrevTabItemClick)
            nextTabItem.Tag = NextPrevValue.Next
            nextTabItem.SvgImage = imageCollection("Next")
            e.Items.Insert(0, nextTabItem)
            Dim prevTabItem As DashboardToolbarItem = New DashboardToolbarItem("Previous tab", AddressOf NextPrevTabItemClick)
            prevTabItem.Tag = NextPrevValue.Prev
            prevTabItem.SvgImage = imageCollection("Prev")
            e.Items.Insert(0, prevTabItem)
            Dim showTabHederItem As DashboardToolbarItem = New DashboardToolbarItem(TabContainer.ShowCaption, "Show tab headers", AddressOf ShowHideTabHeadersItemClick)
            showTabHederItem.SvgImage = imageCollection("ShowCaption")
            e.Items.Insert(0, showTabHederItem)
            Dim slideShowItem As DashboardToolbarItem = New DashboardToolbarItem(slideShowTimer.Enabled, "Slideshow", AddressOf SlideShowItemClick)
            slideShowItem.SvgImage = imageCollection("Slideshow")
            e.Items.Insert(0, slideShowItem)
        End Sub

        Private Sub SlideShowTimerTick(ByVal sender As Object, ByVal e As System.EventArgs)
            transitionManager.StartTransition(dashboardViewer)
            Try
                ShowNextPrevTab(NextPrevValue.Next)
            Finally
                transitionManager.EndTransition()
            End Try
        End Sub

        Private Sub ShowNextPrevTab(ByVal value As NextPrevValue)
            Dim tabContainer As TabContainerDashboardItem = Me.TabContainer
            If tabContainer IsNot Nothing Then
                Dim increment As Integer = If(value = NextPrevValue.Next, 1, -1)
                Dim tabContainerName As String = tabContainer.ComponentName
                Dim selectedIndex As Integer = dashboardViewer.GetSelectedTabPageIndex(tabContainerName)
                Dim pageCount As Integer = tabContainer.TabPages.Count
                dashboardViewer.SetSelectedTabPage(tabContainerName, (selectedIndex + pageCount + increment) Mod pageCount)
            End If
        End Sub

        Private Sub NextPrevTabItemClick(ByVal args As DashboardToolbarItemClickEventArgs)
            ShowNextPrevTab(CType(args.Item.Tag, NextPrevValue))
        End Sub

        Private Sub SlideShowItemClick(ByVal args As DashboardToolbarItemClickEventArgs)
            If args.Item.Checked.HasValue AndAlso args.Item.Checked.Value Then
                slideShowTimer.Start()
            Else
                slideShowTimer.Stop()
            End If
        End Sub

        Private Sub ShowHideTabHeadersItemClick(ByVal args As DashboardToolbarItemClickEventArgs)
            Dim tabContainer As TabContainerDashboardItem = Me.TabContainer
            If tabContainer IsNot Nothing Then
                If args.Item.Checked.HasValue AndAlso args.Item.Checked.Value Then
                    tabContainer.ShowCaption = True
                Else
                    tabContainer.ShowCaption = False
                End If
            End If
        End Sub

        Private Sub DashboardViewerSelectedTabPageChanged(ByVal sender As Object, ByVal e As SelectedTabPageChangedEventArgs)
            Dim tabContainer As TabContainerDashboardItem = Me.TabContainer
            If tabContainer IsNot Nothing AndAlso Not tabContainer.ShowCaption Then
                Dim selectedTabPageName As String = e.Page
                dashboardViewer.Dashboard.Title.Text = Me.TabContainer.TabPages(selectedTabPageName).Name
            End If
        End Sub

        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            If disposing AndAlso components IsNot Nothing Then
                components.Dispose()
            End If

            If disposing Then
                RemoveHandler slideShowTimer.Tick, AddressOf SlideShowTimerTick
                slideShowTimer.Dispose()
            End If

            MyBase.Dispose(disposing)
        End Sub
    End Class

    Friend Enum NextPrevValue
        [Next]
        Prev
    End Enum
End Namespace
